node.go

```diff
@@ -1,11 +1,80 @@
 package pgs
 
+import (
+	"strings"
+
+	"github.com/golang/protobuf/protoc-gen-go/descriptor"
+)
+
 // Node represents any member of the proto descriptor AST. Typically, the
 // highest level Node is the Package.
 type Node interface {
 	accept(Visitor) error
 }
 
+// NodeMeta contains metadata for a node.
+//
+// See the documentation for .google.protobuf.SourceCodeInfo for more details.
+//
+// https://github.com/google/protobuf/blob/master/src/google/protobuf/descriptor.proto
+type NodeMeta struct {
+	*descriptor.SourceCodeInfo_Location
+}
+
+// Line returns the starting line for the node, or 0 if unknown.
+func (n *NodeMeta) Line() int {
+	if n == nil || n.SourceCodeInfo_Location == nil {
+		return 0
+	}
+	if len(n.SourceCodeInfo_Location.Span) < 1 {
+		return 0
+	}
+	return int(n.SourceCodeInfo_Location.Span[0])
+}
+
+// Column returns the starting column for the node, or 0 if unknown.
+func (n *NodeMeta) Column() int {
+	if n == nil || n.SourceCodeInfo_Location == nil {
+		return 0
+	}
+	if len(n.SourceCodeInfo_Location.Span) < 2 {
+		return 0
+	}
+	return int(n.SourceCodeInfo_Location.Span[1])
+}
+
+// TODO: should EndLine and EndColumn return Line and Column by default?
```

Comment:
I'd say yes. EndLine would be equal to Line if it's a single-line span. From descriptor.proto:

> // Always has exactly three or four elements: start line, start column,
> // end line (optional, otherwise assumed same as start line), end column.

EndColumn will/should always be set and will always be the last element: `Span[len(Span)-1]`